ui/src/shared/components/AnnotationWindow.js

```diff
@@ -0,0 +1,43 @@
+import React, {PropTypes} from 'react'
+
+import * as schema from 'shared/schemas'
+
+const containerLeftPadding = 16
+
+const windowDimensions = (anno, dygraph) => {
+  // TODO: export and test this function
+  const [startX, endX] = dygraph.xAxisRange()
+  const startTime = Math.max(+anno.startTime, startX)
+  const endTime = Math.min(+anno.endTime, endX)
+
+  const windowStartXCoord = dygraph.toDomXCoord(startTime)
+  const windowEndXCoord = dygraph.toDomXCoord(endTime)
+
+  const windowWidth = windowEndXCoord - windowStartXCoord
+  const isDurationNegative = windowWidth < 0
+  const foo = isDurationNegative ? windowWidth : 0
```

Comment:
I think we can strive to not have `foo`s. :)

Reply:
oh my god

Reply:
:+1: